codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionHandler.java

```diff
@@ -552,7 +558,9 @@ private ChannelFuture goAway(ChannelHandlerContext ctx, Http2Exception cause) {
         long errorCode = cause != null ? cause.error().code() : NO_ERROR.code();
         ByteBuf debugData = Http2CodecUtil.toByteBuf(ctx, cause);
         int lastKnownStream = connection().remote().lastStreamCreated();
-        return goAway(ctx, lastKnownStream, errorCode, debugData, ctx.newPromise());
+        ChannelFuture future = goAway(ctx, lastKnownStream, errorCode, debugData, ctx.newPromise());
+        ctx.flush();
```

Comment:
The way the code is now we would should be OK to move this into `close(ctx, promise)` right?  However we have identified calling flush multiple times (which it seems like it will be if `onException` calls `onConnectionError` which calls `goAway`) so better safe than sorry for the future I guess? If this is your rational too maybe add a comment so we don't accidentally remove it?